Amplify/Categories/API/Operation/NondeterminsticOperation.swift

```diff
@@ -0,0 +1,99 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+
+import Combine
+/**
+ A non-deterministic operation offers multiple paths to accomplish its task.
+ It attempts the next path if all preceding paths have failed with an error that allows for continuation.
+ */
+enum NondeterminsticOperationError: Error {
+    case totalFailure
+    case cancelled
+}
+
+final class NondeterminsticOperation<T> {
```

Comment:
it would be great to see some generic unit tests that tests this new class

Reply:
The [unit test cases](https://github.com/aws-amplify/amplify-swift/pull/3666/files#diff-9d21da7a8fa50d80141fa292be0f324ed389484cefbe2288e676b40302f6d246R9) are added and collapsed by the github diff tool.